File: svc-md/src/main/java/com/beancounter/marketdata/providers/fxrates/EcbRates.java
package com.beancounter.marketdata.providers.fxrates;

import java.math.BigDecimal;
import java.util.Date;
import java.util.Map;
import lombok.Data;

@Data
public class EcbRates {
  private String base;
  private Date date;
  private Map<String, BigDecimal> rates;

}

File: svc-data/src/main/java/com/beancounter/marketdata/providers/fxrates/EcbRates.kt
package com.beancounter.marketdata.providers.fxrates

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer
import java.math.BigDecimal
import java.time.LocalDate

data class EcbRates constructor(
        val base: String,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
        @JsonSerialize(using = LocalDateSerializer::class)
        @JsonDeserialize(using = LocalDateDeserializer::class)
        val date: LocalDate,
        val rates: Map<String, BigDecimal>

)